pkg/registry/rbac/rest/storage_rbac.go

```diff
@@ -199,6 +201,7 @@ func (p *PolicyData) EnsureRBACPolicy() genericapiserver.PostStartHookFunc {
 				if err != nil {
 					// don't fail on failures, try to create as many as you can
 					utilruntime.HandleError(fmt.Errorf("unable to reconcile clusterrole.%s/%s: %v", rbac.GroupName, clusterRole.Name, err))
+					someReconciliationFailed = true
```

Comment:
If we are going to start from the beginning, I would just `return false, nil` here rather than trying to complete the reconciliations.

Reply:
I am trying to avoid repeating all the initialization logic

Happy to defer on this but it does seem like this will create some unnecessary repeated work and could actually cause starvation. For example with 2 things in the list to process, suppose the first item always fails and second item always succeeds, with your strategy the second item will never be tried.